### PR TITLE
Get planning scene check

### PR DESCRIPTION
**Pull request by davetcoleman**
Fix for error that sometimes occurs when the Rviz Motion Planning plugin is launch with the move_group node in the same launch file. Before making a service call to get an updated planning scene, this causes it to wait for the service to exist up to 5 seconds before showing the error

```
Failed to call service /get_planning_scene
```

The 5 seconds is arbitrary but is enough to allow the timing issue to usually work out. This fix doesn't change any functionality, just helps remove the error message.

### CODE REVIEW

File: planning/planning_scene_monitor/src/planning_scene_monitor.cpp [hunk 1]
```diff
@@ -419,6 +419,13 @@ bool planning_scene_monitor::PlanningSceneMonitor::requestPlanningSceneState(con
       srv.request.components.LINK_PADDING_AND_SCALING |
       srv.request.components.OBJECT_COLORS;
 
+  // Make sure client is connected to server
+  if (!client.exists())
```
Comment (isucan) [inline planning/planning_scene_monitor/src/planning_scene_monitor.cpp:423]: I think this should fail if the service still does not exist after the timeout passes.

**Comment by isucan** [comment]
+1 
If possible, I'd collapse these changes into one commit.

File: planning/planning_scene_monitor/src/planning_scene_monitor.cpp (hunk 1 above)
Comment (davetcoleman) [inline planning/planning_scene_monitor/src/planning_scene_monitor.cpp:423]: It reverts to its current behavior of throwing an error to the console and continuing (code further down)

**Comment by davetcoleman** [comment]
I don't think you can rebase the merge commits. I don't know how to avoid those commits, it's always done that. Isn't that normal? From my research:

> So, long story short: Yes, having merge commits is perfectly fine and you should not worry about them.

http://stackoverflow.com/questions/8509396/git-pull-results-in-extraneous-merge-branch-messages-in-commit-log

**Comment by acornacorn** [comment]
You can get rid of the extra commits like this:

   cd src/moveit_ros
   git checkout get_planning_scene_check
   git fetch origin
   git rebase origin/hydro-devel

and then push it back to the same pull request with

   git push davetcoleman +HEAD:get_planning_scene_check

(The + is necessary to push to the same branch after a rebase.)
This results in a single commit and a much cleaner history.

**Comment by davetcoleman** [comment]
I don't know why, but that rebase command caused this PR to become 23 commits when it was ~5! I have no idea what to do now. I tried rebasing 3 times in different branches.

**Comment by acornacorn** [comment]
Worked for me.  I pushed the result as fix_dave_pr_446 on my fork.  You should be able to get it and fix the pull request with:

git remote add acorn git@github.com:acornacorn/moveit_ros.git
git fetch acorn
git checkout -b rebased_fix acorn/fix_dave_pr_446
 ---  Test and make sure the change is what you want here
git push davetcoleman +HEAD:get_planning_scene_check

-Acorn

P.S. afterwards you can drop my fork from your local repo (if you want) with
git remote rm acorn

**Comment by acornacorn** [comment]
Thought of an easier way to fix this:

  git fetch origin
  git checkout -b cherry_fix origin/hydro-devel
  git cherry-pick a37481021231bb638d9d28ddd813f9ae24c740ae

--- make sure it works and is what you want

  git push davetcoleman +HEAD:get_planning_scene_check

**Comment by davetcoleman** [comment]
Thanks for the tips, my hydro-devel branch was somehow dirty so I deleted it and pulled a new one.

**Pull request closed without merging**